Extensions/PlatformBehavior/platformerobjectruntimebehavior.ts

```diff
@@ -1251,11 +1094,506 @@ namespace gdjs {
+
+      checkTransitionBeforeX() {}
+
+      beforeMovingX() {}
+
+      checkTransitionBeforeY(timeDelta: float) {
+        const behavior = this._behavior;
+        //Go on a ladder
+        behavior.checkTransitionOnLadder();
+        //Jumping
+        behavior.checkTransitionJumping();
+
+        //Grabbing a platform
+        if (
+          behavior._canGrabPlatforms &&
+          behavior._requestedDeltaX !== 0 &&
+          behavior._lastDeltaY >= 0
+        ) {
+          behavior.checkGrabPlatform();
+        }
+      }
+
+      beforeMovingY(timeDelta: float, oldX: float) {
+        const behavior = this._behavior;
+
+        //Fall
+        if (!this._jumpingFirstDelta) {
+          behavior.fall(timeDelta);
+        }
+        this._jumpingFirstDelta = false;
+
+        // Check if the jump key is continuously held since
+        // the beginning of the jump.
+        if (!behavior._jumpKey) {
+          this._jumpKeyHeldSinceJumpStart = false;
+        }
+        this._timeSinceCurrentJumpStart += timeDelta;
+        behavior._requestedDeltaY -= this._currentJumpSpeed * timeDelta;
+
+        // Decrease jump speed after the (optional) jump sustain time is over.
+        const sustainJumpSpeed =
+          this._jumpKeyHeldSinceJumpStart &&
+          this._timeSinceCurrentJumpStart < behavior._jumpSustainTime;
+        if (!sustainJumpSpeed) {
+          this._currentJumpSpeed -= behavior._gravity * timeDelta;
+        }
+        if (this._currentJumpSpeed < 0) {
+          behavior._setFalling();
+        }
+      }
+
+      toString(): String {
+        return 'Jumping';
+      }
+    }
+
+    /**
+     * The object grabbed the edge of a platform and is standing there.
+     */
+    export class GrabbingPlatform implements State {
+      private _behavior: PlatformerObjectRuntimeBehavior;
+      private _grabbedPlatform: any = null;
+      private _grabbedPlatformLastX: any;
+      private _grabbedPlatformLastY: any;
+
+      constructor(behavior: PlatformerObjectRuntimeBehavior) {
+        this._behavior = behavior;
+      }
+
+      enter(grabbedPlatform: PlatformRuntimeBehavior) {
+        this._grabbedPlatform = grabbedPlatform;
+        this._behavior._canJump = true;
+        this._behavior._currentFallSpeed = 0;
+      }
+
+      leave() {
+        this._grabbedPlatform = null;
+      }
+
+      beforeUpdatingObstacles() {}
+
+      checkTransitionBeforeX() {
+        const behavior = this._behavior;
+        //Check that the grabbed platform object still exists and is near the object.
+        if (
+          !behavior._isIn(
+            behavior._potentialCollidingObjects,
+            this._grabbedPlatform.owner.id
+          )
+        ) {
+          behavior._releaseGrabbedPlatform();
+        }
+      }
+
+      beforeMovingX() {
+        const behavior = this._behavior;
+        //Shift the object according to the grabbed platform movement.
+        // this._behavior erases any other movement
```

Comment:
I would rather say "when grabbing a platform, no other movement will be attempted by the behavior"?

Reply:
I did a bad find and replace. It was: "this erases any other movement" 
I think it's to explain why += is not used.